File: KengenWaterApp/KengenWaterApp/app/src/main/java/com/water/alkaline/kengen/data/db/repo/SaveRepo.java
package com.water.alkaline.kengen.data.db.repo;

import android.app.Application;

import com.water.alkaline.kengen.data.db.AppDB;
import com.water.alkaline.kengen.data.db.dao.SaveDao;
import com.water.alkaline.kengen.model.SaveEntity;

import java.util.List;

public class SaveRepo {
    SaveDao dao;

    public SaveRepo(Application application) {
        AppDB database = AppDB.getInstance(application);
        dao = database.saveDao();
    }

    public void insert(SaveEntity entity) {
        dao.insert(entity);
    }
    public void delete(SaveEntity entity) {
        dao.delete(entity);
    }
    public void deleteAll() {
        dao.deleteAll();
    }

    public List<SaveEntity> getAll() {
        return dao.getAll();
    }

    public List<SaveEntity> getbyVideoId(String data) {
        return dao.getbyVideoId(data);
    }
}

File: KengenWaterApp/KengenWaterApp/app/src/main/java/com/water/alkaline/kengen/data/db/repo/SaveRepo.kt
package com.water.alkaline.kengen.data.db.repo

import com.water.alkaline.kengen.data.db.dao.SaveDao
import com.water.alkaline.kengen.model.SaveEntity
import javax.inject.Inject

class SaveRepo @Inject constructor(val dao: SaveDao){

    fun insert(entity: SaveEntity?) {
        dao.insert(entity)
    }

    fun delete(entity: SaveEntity?) {
        dao.delete(entity)
    }

    fun deleteAll() {
        dao.deleteAll()
    }

    val all: List<SaveEntity>
        get() = dao.all

    fun getByVideoId(data: String?): List<SaveEntity> {
        return dao.getbyVideoId(data)
    }
}
